### PR TITLE
[#109] Fix travis cache

### PR DESCRIPTION
**Pull request by willbasky**
Resolves [#109](https://github.com/kowainik/summoner/issues/109)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,12 @@
 # Summoner
 
+1.0.7
+=====
+
+* Add stackage badges
+* Fix travis cache
```
Comment (ChShersh) [inline CHANGELOG.md]: Could you please write changelog according to format? See examples below. Basically, add link to the issue and write short description on the next line.

File: src/Summoner/Template.hs [hunk 2]
```diff
@@ -408,8 +417,8 @@ createStackTemplate ProjectData{..} = Dir (toString repo) $
 
         cache:
           directories:
-          - "$$HOME/.stack"
-          - "$$TRAVIS_BUILD_DIR/.stack-work"
+          $travisCabalCache
+          $travisStackCache
```
Comment (ChShersh) [inline src/Summoner/Template.hs]: Could you also, please, specify cache directories in `.travis.yml` file for `summoner` itself? Now there's only stack but `cabal` should be there as well.

Comment (willbasky) [reply src/Summoner/Template.hs]: Do you mean `$TRAVIS_BUILD_DIR/dist-newstyle`?

Comment (ChShersh) [reply src/Summoner/Template.hs]: @willbasky No, I mean changing this file. Travis configs for `summoner` itself:

* https://github.com/kowainik/summoner/blob/master/.travis.yml

**Comment by ChShersh** [comment]
@willbasky When you open PR for some issue, you should add comment like `Resolves #number_of_the_issue` to your Pull Request first comment body so issue will be closed automatically when PR is merged. It's much more convenient than closing issues manually each time. See example in your previous PR:

* https://github.com/kowainik/summoner/pull/111

**Comment by willbasky** [comment]
I hope the hindsight will work.

File: src/Summoner/Template.hs [hunk 3]
```diff
@@ -398,6 +398,15 @@ createStackTemplate ProjectData{..} = Dir (toString repo) $
                      then installScriptBoth
                      else installScriptCabal
                 else installScriptStack
+            travisCabalCache = emptyIfNot cabal
+                [text|
+                - "$$HOME/.cabal"
+                |]
```
Comment (vrom911) [inline src/Summoner/Template.hs]: I guess no need to use `QuasiQuotes` for this simple text 🙂

File: src/Summoner/Template.hs [hunk 4]
```diff
@@ -398,10 +398,7 @@ createStackTemplate ProjectData{..} = Dir (toString repo) $
                      then installScriptBoth
                      else installScriptCabal
                 else installScriptStack
-            travisCabalCache = emptyIfNot cabal
-                [text|
-                - "$$HOME/.cabal"
-                |]
+            travisCabalCache = emptyIfNot cabal "- \"$$HOME/.cabal\""
```
[the remark this answers is not in the document]
Comment (ChShersh) [reply src/Summoner/Template.hs]: @willbasky Unfortunately, no. But you can create new branch with name like `hot-fix-to-production`

[the remark this answers is not in the document]
Comment (willbasky) [reply src/Summoner/Template.hs]: > It's better to create project each time if some template files are changed.

@ChShersh what projects did you mean?

[the remark this answers is not in the document]
Comment (ChShersh) [reply src/Summoner/Template.hs]: @willbasky I mean test project with the help of `summoner`.

**Pull request merged**